sympy/tensor/index_methods.py

```diff
@@ -23,27 +23,43 @@
 class IndexConformanceException(Exception):
     pass
 
+def _unique_and_repeated(inds):
+    """returns the unique and repeated indices. This is used in the function
```

Comment:
At the top of this file you need to import OrderedDict.

In the docstring, it is not important to say who uses this helper. That will be self-evident from the codebase now and in the future. You probably should indicate that the order of indices is preserved, however, and use a simple example like `[2, 3, 1, 3, 0, 4, 0] -> [2, 1, 4], [3, 0]`.

Reply:
@smichr  Thanks for the review, I have made a few changes. Pls take a look and suggest improvement(s)

Reply:
@smichr  also, can you re-run the Travis build?
I think there was an issue with one test-run, not the code and that's why one of the test is giving an error.